R/list-flatten.R

```diff
@@ -45,11 +46,20 @@ list_flatten <- function(
   ) {
   vec_check_list(x)
 
-  x <- map_if(x, vec_is_list, identity, .else = list)
-  vec_unchop(
-    x,
+  # Take the proxy as we restore on exit
+  proxy <- vec_proxy(x)
+
+  # Unclass S3 lists to avoid their coercion methods. Wrap atoms in a
```

Comment:
Unclassing provides support for strict types that are not coercible to list (as the current `list_of` implementation).

Reply:
Add test of `list_flatten()` applied to `list_of()`?

Reply:
I added a test for a flat `list_of` and another for strict classes (as list-of will be relaxed soon).